src/v2/env_file.rs

```diff
@@ -79,6 +82,7 @@ fn parses_docker_compatible_env_files() {
 # These are environment variables:
 FOO=foo
 BAR=2
+BAZ
```

Comment:
As far as I can tell from the documentation, this is a non-standard extension to the [`.env` format](https://docs.docker.com/compose/compose-file/#env_file):

> Compose expects each line in an env file to be in `VAR=VAL` format. Lines beginning with `#` (i.e. comments) are ignored, as are blank lines.

So unless we can find some evidence that Docker supports this extension, we should remove this code.